File: app/src/main/java/edu/anbu/libraryofalexandria/HomeActivity.java
package edu.anbu.libraryofalexandria;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class HomeActivity extends AppCompatActivity implements BookItemClickListener {

    private BookAdapter bookAdapter;
    private HomeViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        RecyclerView booksRecyclerView = findViewById(R.id.books_recycler_view);
        FloatingActionButton addBookFabButton = findViewById(R.id.add_book_floating_button);

//        addDummyBooks();
        booksRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        bookAdapter = new BookAdapter(this, false);
        booksRecyclerView.setAdapter(bookAdapter);
        booksRecyclerView.addItemDecoration(new DividerItemDecoration(this, LinearLayoutManager.HORIZONTAL));

        viewModel = ViewModelProviders.of(this).get(HomeViewModel.class);

        addBookFabButton.setOnClickListener(view -> startActivity(new Intent(HomeActivity.this, AddBookActivity.class)));
        addObservables();
    }

    private void addObservables() {
        viewModel.getBookList().observe(this, books -> bookAdapter.updateItems(books));
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_home, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == R.id.search_menu_item) {
            startActivity(new Intent(this, SearchBookActivity.class));
        } else if (itemId == R.id.short_by_author_menu_item) {

        } else if (itemId == R.id.short_by_name_menu_item) {

        } else if (itemId == R.id.short_by_publisher_menu_item) {

        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onItemClicked(View v, int position) {
//        Intent intent = new Intent(this, BookDetailActivity.class);
//        intent.putExtra("Book", bookList.get(position));
//        startActivity(intent);
    }

//    private void getAllBook() {
//        new SelectAsyncTask(books -> {
//            bookList.addAll(books);
//            bookAdapter.updateItems();
//        }).execute();
//    }

//    private void addDummyBooks() {
//        Book book1 = new Book("The Java Programming Language", 1, "James Gosling and ken Arnold", 2, "Programming foundation", "genre", "This book an introduce to programming in Oracle's Java programming language, a widely used programming language software platform.", 2017);
//        Book book2 = new Book("The Java Programming Language", 1, "James Gosling and ken Arnold", 2, "Programming foundation", "genre", "This book an introduce to programming in Oracle's Java programming language, a widely used programming language software platform.", 2017);
//        bookList.add(book1);
//        bookList.add(book2);
//    }

//    private static class SelectAsyncTask extends AsyncTask<Void, Void, List<Book>> {
//
//        BookSelectListener listener;
//
//        SelectAsyncTask(BookSelectListener listener) {
//            this.listener = listener;
//        }
//
//        @Override
//        protected List<Book> doInBackground(Void... voids) {
//            return AppDatabase.getInstance().bookDao().getAll();
//        }
//
//        @Override
//        protected void onPostExecute(List<Book> books) {
//            super.onPostExecute(books);
//            Log.e("Book list:", books.toString());
//            listener.onBookFetched(books);
//        }
//    }
}

File: app/src/main/java/edu/anbu/libraryofalexandria/HomeActivity.kt
package edu.anbu.libraryofalexandria

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.floatingactionbutton.FloatingActionButton

class HomeActivity : AppCompatActivity(), BookItemClickListener {
    private lateinit var bookAdapter: BookAdapter
    private lateinit var viewModel: HomeViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)
        val booksRecyclerView = findViewById<RecyclerView>(R.id.books_recycler_view)
        val addBookFabButton = findViewById<FloatingActionButton>(R.id.add_book_floating_button)

        booksRecyclerView.layoutManager = LinearLayoutManager(this)
        bookAdapter = BookAdapter(this, false)
        booksRecyclerView.adapter = bookAdapter
        booksRecyclerView.addItemDecoration(
            DividerItemDecoration(
                this,
                LinearLayoutManager.HORIZONTAL
            )
        )
        viewModel = ViewModelProviders.of(this)[HomeViewModel::class.java]
        addBookFabButton.setOnClickListener {
            startActivity(
                Intent(
                    this@HomeActivity,
                    AddBookActivity::class.java
                )
            )
        }
        addObservables()
    }

    private fun addObservables() {
        viewModel.getBookList()?.observe(this) { books: List<Book> -> bookAdapter.updateItems(books) }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_home, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.search_menu_item -> startActivity(Intent(this, SearchBookActivity::class.java))
            R.id.short_by_author_menu_item -> {}
            R.id.short_by_name_menu_item -> {}
            R.id.short_by_publisher_menu_item -> {}
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onItemClicked(v: View?, position: Int) {
//        Intent intent = new Intent(this, BookDetailActivity.class);
//        intent.putExtra("Book", bookList.get(position));
//        startActivity(intent);
    }
}